portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,34 +32,21 @@
  * and containing them in a JSON string. 
  */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public final class DataServlet extends HttpServlet {
 
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final CommentService commentService = new CommentService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("year", SortDirection.DESCENDING)
-                                      .addSort("month", SortDirection.DESCENDING)
-                                      .addSort("day", SortDirection.DESCENDING);
-
+    
+    Query query = new Query("Comment").addSort("postedTime", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
-
     String amountParam = request.getParameter("comment-amount");
-    int commentAmount = amountParam == null ? 10 : Integer.parseInt(amountParam);
-
-    List<Comment> comments = new ArrayList<>();
-    Iterable<Entity> newComments = results.asIterable(FetchOptions.Builder.withLimit(commentAmount));    
-    
-    for (Entity entity : newComments) {
 
-      String text = (String) entity.getProperty("text");
-      long day = (long) entity.getProperty("day");
-      long month = (long) entity.getProperty("month");
-      long year = (long) entity.getProperty("year");
-      
-      Comment comment = new Comment(text, day, month, year);
-      comments.add(comment);
-    }
+    /** Default value for commentAmount is 10 in case there's a parsing error. */
```

Comment:
nit: Forgot to mention, in-line comments are typically represented with "//" instead of the "/**" javadoc ones.

See https://google.github.io/styleguide/javaguide.html#s4.8.6-comments

Reply:
Done.